tests/test_files.py

```diff
@@ -11,16 +11,21 @@
 
 
 class Base(BaseClass):
-    @StoreFiles.arg_references({"target_file": 2})
+    @StoreFiles.arg_references(
+        {"target_file": 2}, cassette=PersistentObjectStorage().cassette
+    )
     def create_file_content(self, value, target_file):
         with open(target_file, "w") as fd:
             fd.write(value)
         return "value"
 
-    @StoreFiles.arg_references({"target_dir": 2})
+    @StoreFiles.arg_references(
+        {"target_dir": 2}, cassette=PersistentObjectStorage().cassette
```

Comment:
I don't understand this syntax (I can think more and dig in the code, but others can have same problems as well):

- I would like to read decorators/calls as sentence: `StoreFiles.where_arg_references` ...
- Do we need that dict here? Cannot we use `StoreFiles.where_arg_references(target_dir=2, cassette=PersistentObjectStorage().cassette)`?
- What is that `2` in `"target_dir": 2`?
- We are using a global cassette here? (`cassette=PersistentObjectStorage().cassette`)
  - We can make it more accessible -- e.g. `from requre import global_cassette`

Reply:
it is a dictionary containing parameters and position as parameter of function.
could be also changed to `target_dir=2` as params, but not sure if this will be more easy to read the function code:
eg. you have function definition:
```
def some(target_dir='some/path", random_file="/another/path", other_param=something):
...
```
it leads to decorator `StoreFiles.where_arg_references(file_params={"target_dir"=0, "random_file"=1}...)`
in your way, to be defined as **kwargs of `where_arg_references` so will not be easy to read, how to write these params, but then it will be possible to write 
```
StoreFiles.where_arg_references(..., target_dir=0, random_file=1)
```
and then it will internally use same logic as there.

In one implementation I tried also to use inspect.getfullargsspec to try to find positions of params, but I've found it is very dependent how you use decorators more decorators, because  decorated fucntions leads to *args and **kwargs, and then I were unable to dereference names. Not sure if I described it well :-)

Reply:
My point was not that you need to describe that here, but to potential users...;) I was more speaking about the syntax => how to set that information more clearly.

But now I see it's a bit complicated situation. (For me, the most magical thing is the number.)

Some wild ideas if we can't easily get rid of the number:

- You are right. This mixes the arguments and `cassette` and still don't help with numbers.
```python
@StoreFiles.where(random_file=1, cassette="asdasd")
def test():
  pass
```

- Have a vector with the argument names on the non-`None` names. 
```python
@StoreFiles.where(arguments=[None, "second"], cassette="asdasd")
def test():
  pass
```

- Have a list of tuples.
```python
@StoreFiles.where(arguments_and_positions=[("second",1), ("first", 0)], storage_file="asdasd")
def test():
  pass
```

- Or, just rename the argument:
```python
@StoreFiles.where(arguments_and_positions={"target_dir": 2}, storage_file="asdasd")
def test():
  pass
```